calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,150 @@
+
+from logutils import configure_logger
 from pycalico import netns
 from pycalico.ipam import IPAMClient, SequentialAssignment
 from pycalico.netns import Namespace
-from pycalico.datastore_datatypes import Rules
+from pycalico.datastore_datatypes import Rules, IPPool
 from pycalico.datastore import IF_PREFIX, DatastoreClient
 from pycalico.datastore_errors import PoolNotFound
 
-print_stderr = functools.partial(print, file=sys.stderr)
+_log = logging.getLogger(__name__)
 
-
-input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
-input_json = json.loads(input_)
-
-# Append to existing env, to avoid losing PATH etc.
-# TODO-PAT: This shouldn't be hardcoded
-env = os.environ.copy()
 ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
-env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
-
-# PROFILE_LABEL = 'CALICO_PROFILE'
-# ETCD_PROFILE_PATH = '/calico/'
 
 ORCHESTRATOR_ID = "rkt"
 HOSTNAME = socket.gethostname()
-NETNS_ROOT= '/var/lib/rkt/pods/run'
+NETNS_ROOT = '/var/lib/rkt/pods/run'
 
-def main():
+def main(env, conf_in):
     mode = env['CNI_COMMAND']
 
-    if mode == 'init':
-        print_stderr('No initialization work to perform')
-    elif mode == 'ADD':
-        print_stderr('Executing Calico pod-creation plugin')
-        create(
-            container_id=env['CNI_CONTAINERID']
-            )
+    if mode == 'ADD':
+        create(env=env, conf_in=conf_in)
     elif mode == 'DEL':
-        print_stderr('Executing Calico pod-deletion plugin')
-        delete(
-            container_id=env['CNI_CONTAINERID']
-            )
+        delete(env=env, conf_in=conf_in)
 
-def create(container_id):
+def create(env, conf_in):
     """"Handle rkt pod-create event."""
-    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    container_id = env['CNI_CONTAINERID']
+
+    _log.info('Configuring pod %s' % container_id)
     netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
     _datastore_client = IPAMClient()
 
     try:
         endpoint, ip = _create_calico_endpoint(container_id=container_id,
-                                            netns_path=netns_path,
-                                            client=_datastore_client)
+                                               netns_path=netns_path,
+                                               client=_datastore_client,
+                                               conf_in=conf_in,
+                                               interface = env['CNI_IFNAME'])
 
-        _create_profile(endpoint=endpoint, 
-                        profile_name=input_json['name'], 
+        _set_profile_on_endpoint(endpoint=endpoint,
+                        profile_name=conf_in['name'],
                         ip=ip,
                         client=_datastore_client)
     except CalledProcessError as e:
-        print_stderr('Error code %d creating pod networking: %s\n%s' % (
+        _log.error('ERROR: error code %d creating pod networking: %s\n%s' % (
```

Comment:
Your logger format already has ` %(levelname)s` in the header, so it will automatically add the "ERROR" prefix to this message.